protocol/src/main/java/org/geysermc/mcprotocollib/protocol/codec/MinecraftCodecHelper.java

```diff
@@ -456,82 +586,75 @@ public void writeParticle(ByteBuf buf, Particle particle) {
-            default -> null;
-        };
+                return new DustColorTransitionParticleData(red, green, blue, scale, newRed, newGreen, newBlue);
+            case ENTITY_EFFECT:
+                return new EntityEffectParticleData(buf.readInt());
+            case ITEM:
+                return new ItemParticleData(this.readOptionalItemStack(buf));
+            case SCULK_CHARGE:
+                return new SculkChargeParticleData(buf.readFloat());
+            case SHRIEK:
+                return new ShriekParticleData(this.readVarInt(buf));
+            case VIBRATION:
+                return new VibrationParticleData(this.readPositionSource(buf), this.readVarInt(buf));
+            default:
+                return null;
+        }
     }
 
     public void writeParticleData(ByteBuf buf, ParticleType type, ParticleData data) {
         switch (type) {
-            case BLOCK, BLOCK_MARKER -> {
-                BlockParticleData block = (BlockParticleData) data;
-                this.writeVarInt(buf, block.getBlockState());
-            }
-            case DUST -> {
-                DustParticleData dust = (DustParticleData) data;
-                buf.writeFloat(dust.getRed());
-                buf.writeFloat(dust.getGreen());
-                buf.writeFloat(dust.getBlue());
-                buf.writeFloat(dust.getScale());
-            }
-            case DUST_COLOR_TRANSITION -> {
-                DustColorTransitionParticleData dust = (DustColorTransitionParticleData) data;
-                buf.writeFloat(dust.getRed());
-                buf.writeFloat(dust.getGreen());
-                buf.writeFloat(dust.getBlue());
-                buf.writeFloat(dust.getScale());
-
-                buf.writeFloat(dust.getNewRed());
-                buf.writeFloat(dust.getNewGreen());
-                buf.writeFloat(dust.getNewBlue());
-            }
-            case FALLING_DUST -> {
-                FallingDustParticleData fallingDust = (FallingDustParticleData) data;
-                this.writeVarInt(buf, fallingDust.getBlockState());
-            }
-            case ITEM -> {
-                ItemParticleData item = (ItemParticleData) data;
-                this.writeItemStack(buf, item.getItemStack());
-            }
-            case SCULK_CHARGE -> {
-                SculkChargeParticleData sculkCharge = (SculkChargeParticleData) data;
-                buf.writeFloat(sculkCharge.getRoll());
-            }
-            case SHRIEK -> {
-                ShriekParticleData shriek = (ShriekParticleData) data;
-                this.writeVarInt(buf, shriek.getDelay());
-            }
-            case VIBRATION -> {
-                VibrationParticleData vibration = (VibrationParticleData) data;
-                this.writePositionSource(buf, vibration.getPositionSource());
-                this.writeVarInt(buf, vibration.getArrivalTicks());
-            }
+            case BLOCK:
```

Comment:
This seems like a regression, since we're moving from new-style switch blocks to old-style switch blocks.